shapely/coordinates.py

```diff
@@ -129,6 +141,116 @@ def transform(
     return result
 
 
+def transform_resize(
+    geom: Optional[shapely.Geometry],
+    transformation,
+    include_z: Optional[bool] = False,
+    interleaved: bool = True,
+):
+    """Apply a function to the coordinates in a Geometry object, optionally
+    changing the number of coordinate pairs.
+
+    The transformation function is applied per coordiante sequence. For polygons this
+    means: per ring. For collections this means: per element.
+
+    This function differs with `transform` in the following ways:
+
+    - It only accepts scalar Geometry objects, not arrays.
+    - The number of coordinates per coordinate sequence is allowed to change.
+
+    We recommend only using this function only if the number of coordinates in your
```

Comment:
maybe state the reason? "Because this function is expected to be slower than `transform`, ...."